ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/AttestationValidator.java

```diff
@@ -186,6 +186,8 @@ private InternalValidationResult singleAttestationChecks(final Attestation attes
               if (!is_valid_indexed_attestation(state, indexedAttestation).isSuccessful()) {
```

Comment:
Would is make sense to pass the `ValidatableAttestation` to a new version of `is_valid_indexed_attestation`?  This new method could return early if the `isValidIndexedAttestation` is already set, and otherwise delegate to the existing method that accepts the plain attestation, then set `isValidIndexedAttestation`.

Reply:
Makes sense.